Ray.jl/src/runtime.jl

```diff
@@ -108,16 +108,33 @@ function init(runtime_env::Union{RuntimeEnv,Nothing}=nothing;
     return nothing
 end
 
-# this could go in JLL but if/when global worker is hosted here it's better to
-# keep it local
-get_current_job_id() = ray_jll.ToInt(ray_jll.GetCurrentJobId())
+const CORE_WORKER = Ref{CxxRef{ray_jll.CoreWorker}}()
+
+function GetCoreWorker()
+    return ray_jll.GetCoreWorker()
+    # if !isassigned(CORE_WORKER)
+    #     CORE_WORKER[] = ray_jll.GetCoreWorker()
+    # end
+    # return CORE_WORKER[]
+end
```

Comment:
Using the write-once version of this function results in most of the Ray.jl test suite failing with:
```
C++ object of type N3ray4core10CoreWorkerE was deleted
```
I'll try a couple other things

Reply:
Interesting. If I change our global reference to be a `Ref{ray_jll.CoreWorker}` then things work fine. It seems that the `CxxRef` doesn't hold onto the allocated memory.